### DISCUSSION TITLE
Simplify cf_ai / cf_recruited

### DISCUSSION THREAD

**Discussion by FrozenEmpire** [ChilliHugger/The-Lords-Of-Midnight#180] [Code Algorithm Questions] [answered]
An observation/potential simplification

Isn't cf_ai always the opposite of cf_recruited ?
In other words, cf_ai could be removed, and IsAIControlled() replaced with IsRecruited() ?

So code like this in mxengine.cpp becomes largely redundant ?
(except check on Morkin getting recruited could be moved to recruitment function...)

if ( !morkin->IsRecruited() )
    morkin->Flags().Set(cf_ai);

// fix for morkin being AI character
// after being recruited
if ( morkin->IsRecruited() && morkin->IsAIControlled() ) {
    morkin->Flags().Reset(cf_ai);
    if ( morkin->IsInTunnel() )
        morkin->looking=DR_NORTH;
}

**Comment by IcemarkUK** [comment, marked as answer]
Yes, technically that's true, when you consider the original. I intentionally added AI as a separate flag because I envisaged a character who was "recruited" ie: controlled by the player, but could also be AI. ie: Make decisions for themselves. This is more likely to happen in The Citadel. We could say that this is still all in flux.

**Reply by FrozenEmpire** [reply]
Interesting, I hadn’t considered The Citadel ;)
